src/Compilers/Test/Core/TestableCompiler.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.VisualBasic;
+
+namespace Roslyn.Test.Utilities
+{
+    internal record struct TestableCompilerFile(string FilePath, TestableFile TestableFile)
```

Comment:
nit: we haven't really used records in the compiler codebase yet because we didn't really need the structural equality part. But this in tests, so I think I'm okay with using records for brevity.

Reply:
There are some decently heavy use of `record` in the rebuild code base. But you are right I don't need structural equality here so reverting to plain `struct`